scene/gui/text_edit.cpp

```diff
@@ -5644,6 +5660,35 @@ void TextEdit::_scroll_lines_down() {
 }
 
 // Minimap
+
+void TextEdit::_update_minimap_hover() {
+	const Point2 mp = get_local_mouse_pos();
+	const int xmargin_end = get_size().width - style_normal->get_margin(SIDE_RIGHT);
+
+	const bool hovering_sidebar = mp.x > xmargin_end - minimap_width && mp.x < xmargin_end;
+	if (!hovering_sidebar) {
+		if (hovering_minimap) {
+			// Only redraw if the hovering status changed.
+			hovering_minimap = false;
+			print_line("no longer hovering");
+			update();
+		}
+
+		// Return early to avoid running the operations below when not needed.
+		return;
+	}
+
+	const int row = get_minimap_line_at_pos(Point2i(mp.x, mp.y));
+
+	const bool new_hovering_minimap = row >= get_first_visible_line() && row < get_last_full_visible_line();
```

Comment:
This seems to need `row <= get_last_full_visible_line()` to handle hovering past the end of the script.